### PR TITLE
feat: `client_id` auth

### PR DESCRIPTION
**Pull request by HarryET**
# Description
Adds non-breaking client id/jwt validation on the client endpoints

## How Has This Been Tested?
N/a, non-breaking so will do integration testing with SDK teams once they have upgraded

## Due Diligence

* [ ] Breaking change
* [ ] Requires a documentation update
* [ ] Requires a e2e/integration test update

### CODE REVIEW

File: terraform/variables.tf [hunk 1]
```diff
@@ -10,7 +10,7 @@ variable "azs" {
 
 variable "public_url" {
   type    = string
-  default = "echo.walletconnect.com"
+  default = "https://echo.walletconnect.com"
```
Comment (HarryET) [inline terraform/variables.tf]: @Rakowskiii I think its the `https://` prefix here causing some of the terraform issues

File: src/handlers/delete_client.rs [hunk 2]
```diff
@@ -1,23 +1,39 @@
...
     Path((tenant_id, id)): Path<(String, String)>,
     StateExtractor(state): StateExtractor<Arc<AppState>>,
+    headers: HeaderMap,
 ) -> Result<Response> {
     let id = id
         .trim_start_matches(DECENTRALIZED_IDENTIFIER_PREFIX)
         .to_string();
 
+    let client_to_be_deleted = ClientId::new(id.clone().into());
+    if !authenticate_client(headers, &state.config.public_url, |client_id| {
```
Comment (HarryET) [inline src/handlers/delete_client.rs:28]: `public_url` needs the `https://` or `http://` prefix now. Same for all other function calls

Comment (Rakowskiii) [reply src/handlers/delete_client.rs:28]: Wouldn't this line do that? https://github.com/WalletConnect/echo-server/blob/8f592e4253bee4cb34cf4de7563e80da2338f32f/terraform/ecs/main.tf#L73

Comment (HarryET) [reply src/handlers/delete_client.rs:28]: Yeah thats good!

**Pull request merged**